pkg/log/logshipper/logshipper.go

```diff
@@ -64,24 +64,30 @@ func (ls *LogShipper) Ping() {
 	ls.sender.authtoken = string(token)
 
 	shouldEnable := ls.knapsack.LogIngestServerURL() != ""
-	parsedUrl, err := url.Parse(ls.knapsack.LogIngestServerURL())
-	if err != nil || parsedUrl.String() == "" {
-		// If we have a bad endpoint, just disable for now.
-		// It will get renabled when control server sends a
-		// valid endpoint.
-		shouldEnable = false
-		level.Debug(ls.baseLogger).Log(
-			"msg", "error parsing log ingest server url, shipping disabled",
-			"err", err,
-			"log_ingest_url", ls.knapsack.LogIngestServerURL(),
-		)
+
+	// Only attempt to parse a URL if we have one
+	if shouldEnable {
+		parsedUrl, err := url.Parse(ls.knapsack.LogIngestServerURL())
```

Comment:
I'm surprised you need this. 

https://go.dev/play/p/P2NKDl7t9U_q makes it look like parsing `""` will return empty url object. But I guess I'm not sure how to best handle that. So maybe this is good enough

Reply:
It will return an empty URL object (and will therefore not panic on parsedUrl.String), but I figured no reason to perform the parse if we don't need to

Reply:
Simplified this